src/app/components/profile/profile.component.ts

```diff
@@ -91,26 +86,30 @@ export class ProfileComponent implements OnInit {
       this.processUserDetails();
     });
   }
-
   /**
    * Process user details function.
    */
   processUserDetails() {
+    let first_name = (this.user['first_name'] == '-') ? '' : this.user['first_name'];
+    let last_name = (this.user['last_name'] == '-') ? '' : this.user['last_name'];
+    let affiliation = (this.user['affiliation'] == '-') ? '' : this.user['affiliation'];
+    let google_scholar_url = (this.user['google_scholar_url'] == '-') ? '' : this.user['google_scholar_url'];
+    let github_url = (this.user['github_url'] == '-') ? '' : this.user['github_url'];
+    let linkedin_url = (this.user['linkedin_url'] == '-') ? '' : this.user['linkedin_url'];
```

Comment:
I think you should move all these a little down as upto here hyphen is not assigned to the `user object`. Even I would suggest to move all these inside `updateDetails` function as these variables are not gonna used again.

Reply:
little down?

Reply:
Well, I'm suggesting to move all these inside `updateUserDetails` function.